core/src/main/java/com/linecorp/armeria/server/encoding/EncodingServiceBuilder.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.encoding;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableList;
+
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.server.HttpService;
+
+/**
+ * Builds a new {@link EncodingService} or its decorator function.
+ */
+public final class EncodingServiceBuilder {
+
+    // TODO(minwoox) consider this condition to align with the default text preveiwer.
+    private static final Predicate<MediaType> defaultEncodableContentTypePredicate =
+            contentType -> Stream.of(MediaType.ANY_TEXT_TYPE,
+                                     MediaType.APPLICATION_XML_UTF_8,
+                                     MediaType.JAVASCRIPT_UTF_8,
+                                     MediaType.JSON_UTF_8)
+                                 .anyMatch(contentType::is);
```

Comment:
Could create a `MediaTypeSet` and reuse it, instead of creating a new stream list again and again.

Reply:
Let me address this once #2444 is merged. 😉

Reply:
Oops realized that the `MediaTypeSet` cannot have a media type which has wildcard. So just use Set and iterate on it.

Reply:
Oh no! OK.